app/routes.py

```diff
@@ -1,2 +1,272 @@
-from flask import Blueprint
+from flask import Blueprint, jsonify, make_response, request
+from app.models.task import Task
+from app.models.goal import Goal
+from app import db
+from datetime import datetime, timezone
+import requests
+import os
 
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+def is_input_valid(number):
+    try:
+        int(number)
+    except:
+        return make_response(f"{number} is not an int!", 400)
+
+
+def is_parameter_found(model, parameter_id):
+    if is_input_valid(parameter_id):
+        return is_input_valid(parameter_id)
+    elif model.query.get(parameter_id) is None:
+        return make_response(f"{parameter_id} was not found!", 404)
+    
+
+
+#TASK ROUTES
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return jsonify({"details": "Invalid data"}),400
+    new_task = Task(title=request_body["title"],
+                    description=request_body["description"], 
+                    completed_at=request_body["completed_at"]
+                    )
+
+    db.session.add(new_task)
+    db.session.commit()
+    response_body = {}
+    response_body["task"] = new_task.to_dict()
+    return jsonify(response_body), 201
+
+
+@tasks_bp.route("", methods=["GET"])
+def read_tasks():
+    tasks = Task.query.all()
+    response_body = []
+    sort_query = request.args.get("sort")
+
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+        
+    for task in tasks:
+        response_body.append(
+            task.to_dict())
+    return jsonify(response_body), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_task(task_id):
+    check_not_found = is_parameter_found(Task, task_id)
+    if check_not_found:
+        return check_not_found
+    task = Task.query.get(task_id)
```

Comment:
The use of this helper function is a little bit hard to understand. In addition, the implementation requires that we call `is_input_valid` multiple times. 

Consider refactoring to make the helper function `is_input_valid` and `is_parameter_found` a single function `get_model` that gets a single Task instance or raises an exception. Then in your route function, you can appropriately handle the exception by returning a 400 or 404.